src/weather_man.js

```diff
@@ -8,23 +8,35 @@ class WeatherMan extends React.Component {
     super(props)
     this.state = {
       city: "",
-      weather: "",
+      weatherForecast: "",
     }
+    this.registerCity = this.registerCity.bind(this);
+    this.updateWeatherForecast = this.updateWeatherForecast.bind(this);
   }
 
-  componentDidMount() {
-    getDataFor(this.props.city)
+  registerCity(event) {
+    this.setState({
+      city: event.target.value,
+    })
+  }
+
+  updateWeatherForecast() {
+    getDataFor(this.state.city)
       .then(data => {
 	this.setState({
```

Comment:
When I cloned the code, the indentation looked a bit strange here. I would expect something like this:

```
 updateWeatherForecast() {
   getDataFor(this.state.city)
     .then(data => {
       this.setState({
         weatherForecast: "The weather in " + this.state.city + " now is: " + data.weather[0].description.toUpperCase(),
       })
     })
 }
```

Reply:
My code is exactly like this, not sure what happens when you clone it